pkg/operator/storageclasscontroller/storageclasscontroller.go

```diff
@@ -58,7 +62,7 @@ func (c *StorageClassController) Sync(ctx context.Context, connection *vclib.VSp
 			klog.Errorf("error syncing storage policy: %v", syncResult.Reason)
 			clusterCondition := "storage_class_sync_failed"
 			utils.InstallErrorMetric.WithLabelValues(string(syncResult.CheckStatus), clusterCondition).Set(1)
-			return syncResult, checks.ClusterCheckAllGood
+			return syncResult, checks.ClusterCheckDegrade
```

Comment:
That is intentional. We chose not to degrade clusters if we can't create storage policy for now, so as to allow customers to create their own policies and storage classes. This may change in future though.

Reply:
Thank you for explanation, I've changed the patch to match the desired behavior and I think this PR ready for review.